website/docs/r/nsxt_edgegateway_l2_vpn_tunnel.html.markdown

```diff
@@ -0,0 +1,138 @@
+
+  name        = "client-session"
+  description = "example description"
+
+  session_mode = "CLIENT"
+  enabled      = true
+
+  # must be sub-allocated on the Edge Gateway
+  local_endpoint_ip  = "101.22.30.3"
+  remote_endpoint_ip = "1.2.2.3"
+
+  stretched_network {
+    network_id = data.vcd_routed_network_v2.test_network_client.id
+    # CLIENT sessions need to define a tunnel ID for every stretched network
+    tunnel_id = 1
+  }
+
+  stretched_network {
+    network_id = data.vcd_routed_network_v2.test_network_client_other.id
+    tunnel_id  = 2
+  }
+
+  # Be aware, that if there are changes in the `server-session`, the peer_code
+  # will be updated aswell, so `terraform apply` needs to be run twice
+  peer_code = vcd_nsxt_edgegateway_l2_vpn_tunnel.server-session.peer_code
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `org` - (Optional) The name of organization to use, optional if defined at 
+  provider level. Useful when connected as sysadmin working across different organisations
+* `edge_gateway_id` - (Required) The ID of the edge gateway (NSX-T only). 
+  Can be looked up using `vcd_nsxt_edgegateway` datasource
+* `name` - (Required) The name of the tunnel.
+* `description` - (Optional) The description of the tunnel.
+* `session_mode` - (Required) Mode of the tunnel session (SERVER or CLIENT)
+* `enabled` - (Optional) State of the `SERVER` session, always set to `true` for `CLIENT` 
+  sessions.
+* `connector_initiator_mode` - (Required for `SERVER` sessions) Mode in which 
+  the connection is formed. Only relevant to `SERVER` sessions. One of:
+	* `INITIATOR` - Local endpoint initiates tunnel setup and will also respond to 
+  incoming tunnel setup requests from the peer gateway.
+	* `RESPOND_ONLY` - Local endpoint shall only respond to incoming tunnel setup 
+  requests, it shall not initiate the tunnel setup.
+	* `ON_DEMAND` - In this mode local endpoint will initiate tunnel creation once 
+  first packet matching the policy rule is received, and will also respond to 
+  incoming initiation requests.
+* `local_endpoint_ip` - (Required) The IP address corresponding to the Edge 
+  Gateway the tunnel is being configured on. The IP must be sub-allocated 
+  on the Edge Gateway.
+* `remote_endpoint_ip` - (Required) The IP address of the remote endpoint, which 
+corresponds to the device on the remote site terminating the VPN tunnel.
+* `tunnel_interface` - (Optional) The network CIDR block over which the session 
+  interfaces. Relevant only for SERVER session modes. If not provided, Cloud 
+  Director will attempt to automatically allocate a tunnel interface.
+* `pre_shared_key` - (Required for `SERVER` sessions) The key that is used for 
+  authenticating the connection, only needed for `SERVER` sessions.
+* `peer_code` - (Optional) Encoded string that contains the whole configuration 
+  of a `SERVER` session, including the pre-shared key, so it is user's 
+  responsibility to secure it.
```

Comment:
We should say where we can get this information from, and whether we need to encode it somehow (the resource description mentions Base64